libraries/node-core-library/src/Import.ts

```diff
@@ -264,21 +263,26 @@ export class Import {
    * and a system module is found, then its name is returned without any file path.
    */
   public static resolvePackage(options: IImportResolvePackageOptions): string {
-    const { packageName } = options;
+    const { packageName, includeSystemModules, baseFolderPath, allowSelfReference } = options;
 
-    if (options.includeSystemModules && Import._builtInModules.has(packageName)) {
+    if (includeSystemModules && Import._builtInModules.has(packageName)) {
       return packageName;
     }
 
-    const normalizedRootPath: string = FileSystem.getRealPath(options.baseFolderPath);
+    const normalizedRootPath: string = FileSystem.getRealPath(baseFolderPath);
 
-    if (options.allowSelfReference) {
-      const ownPackage: IPackageDescriptor | undefined = Import._getPackageName(options.baseFolderPath);
+    if (allowSelfReference) {
+      const ownPackage: IPackageDescriptor | undefined = Import._getPackageName(baseFolderPath);
       if (ownPackage && ownPackage.packageName === packageName) {
         return ownPackage.packageRootPath;
       }
     }
 
+    const { error } = PackageName.tryParse(packageName);
+    if (error) {
+      throw new Error(error);
```

Comment:
I'm surprised there isn't a `PackageName.parse` method which would just do this.

Reply:
Lol actually there is. Switched to that.